src/testnet/helium/event_dispatcher.rs

```diff
@@ -0,0 +1,242 @@
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
+use std::thread::spawn;
+use std::net::{SocketAddr};
+use mio::tcp::TcpStream;
+use serde_json::json;
+use serde::Serialize;
+
+use vm::types::{Value, QualifiedContractIdentifier, AssetIdentifier};
+use burnchains::Txid;
+use chainstate::stacks::StacksBlock;
+use chainstate::stacks::events::{StacksTransactionReceipt, StacksTransactionEvent, STXEventType, FTEventType, NFTEventType};
+use chainstate::stacks::db::StacksHeaderInfo;
+use net::StacksMessageCodec;
+
+use super::config::{EventObserverConfig, EventKeyType};
+
+#[derive(Debug)]
+struct EventObserver {
+    sock_addr: SocketAddr
+}
+
+impl EventObserver {
+
+    pub fn new(address: &str, port: u16) -> EventObserver {
+        let sock_addr = SocketAddr::new(address.parse().unwrap(), port);
+        EventObserver { sock_addr }
+    }
+
+    pub fn send(&mut self, filtered_events: Vec<&(Txid, &StacksTransactionEvent)>, chain_tip: &StacksBlock, chain_tip_info: &StacksHeaderInfo, receipts: &Vec<StacksTransactionReceipt>) {
+        // Initiate a tcp socket
+        let stream = TcpStream::connect(&self.sock_addr).unwrap();
```

Comment:
Why are we re-connecting every time, instead of once when the `EventObserver` is instantiated?

Reply:
Also, if the observer crashes, this will cause the node to panic due to the `unwrap()`.

Reply:
Either way, you should consider using `mio::net::TcpStream` here instead, since you'll want to do things like set a connection timeout, activating the no-linger flag, disabling Nagle's algorithm, and so on (but note that `mio::net::TcpStream` is _non-blocking_, so you'll want to (a) create the `std::net::TcpStream`, (b) connect with it, (c) wrap it with `mio::net::Tcpstream`, (d) set the socket options, and (3) revert it to blocking mode (if that's the intended behavior).

Reply:
Can we address this in a separate PR with https://github.com/blockstack/stacks-blockchain/issues/1352 once your HTTP client is available in master?

Reply:
Sure; just make sure there's an issue for it :)